bridge-fungible-token/src/utils.sw

```diff
@@ -63,12 +63,42 @@ fn shift_decimals_left(bn: U256, d: u8) -> Result<U256, BridgeFungibleTokenError
     Result::Ok(bn_clone)
 }
 
+fn shift_decimals_right(bn: U256, d: u8) -> Result<(U256, u32), BridgeFungibleTokenError> {
+    let mut bn_clone = bn;
+    let mut decimals_to_shift: u32 = asm(r1: d) { r1: u32 };
+    let mut r = 0u32;
+
+    // the zero case
+    if (decimals_to_shift == 0u32) {
+        return Result::Ok((bn, 0u32));
+    };
+
+    // the too large case
+    // (there are only 78 decimal digits in a 256bit number)
+    if (decimals_to_shift > 77u32) {
+        return Result::Err(BridgeFungibleTokenError::OverflowError);
+    };
+
+    // math time
+    while (decimals_to_shift > 0u32) {
+        if (decimals_to_shift < 20u32) {
```

Comment:
We can only shift by at most 9 decimals with a u32. I'm also thinking we should not include the max case (9) in this `if` and instead have it handled in the `else` case which precalculates 10**9 (1,000,000,000). This would save some gas in what is probably the most common scenario (18 to 9 decimals) vs dynamically calculating a pow with `decimals_to_shift`. 

*note: we should probably also apply this same rearrangement to `shift_decimals_left` by using `if (decimals_to_shift < 19u32) {`
```suggestion
        if (decimals_to_shift < 9u32) {
```